rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbCreateAlarmNode.java

```diff
@@ -160,4 +162,22 @@ private Alarm buildAlarm(TbMsg msg, JsonNode details, TenantId tenantId) {
                 .build();
     }
 
+    private AlarmSeverity processAlarmSeverity(TbMsg msg) {
+        AlarmSeverity alarmSeverity;
+        if(this.config.isDynamicSeverity()) {
+            String processPattern = TbNodeUtils.processPattern(this.config.getSeverity(), msg);
+            if(!EnumUtils.isValidEnum(AlarmSeverity.class, processPattern)) {
```

Comment:
This check is redundant. The EnumUtils.getEnum will simply return null if the value is not correct.